code/game/objects/items/weapons/shards.dm

```diff
@@ -87,6 +87,7 @@
 				H.Weaken(3)
 				if(affecting.take_damage(5, 0))
 					H.UpdateDamageIcon()
+					affecting.embed(src)
```

Comment:
I will put into its own PR.

However, unless merged, if you are ejected with a glass shard and are barefoot, it will do damage to you with each turf you travel up to the damage cap of your mob.